### PR TITLE
Overflow slashing calculation fix

### PR DESCRIPTION
**Pull request by shayzluf**
[Part of] #2969

---

# Align process slashing to 0.8.1 spec 

https://github.com/ethereum/eth2.0-specs/pull/1286

### CODE REVIEW

File: beacon-chain/core/epoch/epoch_processing_test.go [hunk 1]
```diff
@@ -751,9 +751,9 @@ func TestProcessSlashings_SlashedLess(t *testing.T) {
 				Balances:  []uint64{params.BeaconConfig().MaxEffectiveBalance, params.BeaconConfig().MaxEffectiveBalance},
 				Slashings: []uint64{0, 1e9},
 			},
-			// penality   = validator.effective_balance * min(sum(state.slashings) * 3, total_balance)   // total_balance
-			// 1500000000 = (32 * 1e9)                 * min(sum([0, 1e9])        * 3, ((32 * 1e9) * 2)) // ((32 * 1e9) * 2)
-			want: uint64(30500000000), // 32 * 1e9 - 1500000000
+			// penalty    = validator balance / increment * (3*total_penalties) / total_balance * increment
+			// 1000000000 = (32 * 1e9)        / (1 * 1e9) * (3*1e9)             / (64*1e9)      * (1 * 1e9)
```
Comment (prestonvanloon) [inline beacon-chain/core/epoch/epoch_processing_test.go]: Please update these equations. For example, this still evaluates to 1500000000.0 in python

```python
>>> (32 * 1e9)        / (1 * 1e9) * (3*1e9)             / (64*1e9)      * (1 * 1e9)
1500000000.0
```

Comment (shayzluf) [reply beacon-chain/core/epoch/epoch_processing_test.go]: while using uint

**Comment by prestonvanloon** [comment]
Looking at this again, did we expect the outputs of the slashing to change? Spec tests are failing, but I'm not sure if this would be resolved with updated spec tests for 0.8.1?

**Comment by terencechain** [comment]
> Looking at this again, did we expect the outputs of the slashing to change? Spec tests are failing, but I'm not sure if this would be resolved with updated spec tests for 0.8.1?

No, I don't think the outputs of slashing is suppose to change

**Pull request merged**